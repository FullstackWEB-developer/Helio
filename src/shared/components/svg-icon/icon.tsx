export enum Icon {
    Add = 'Add',
    AddBlack = 'AddBlack',
    AddContact = 'AddContact',
    Alert = 'Alert',
    ArrowDown = 'ArrowDown',
    ArrowDownward = 'ArrowDownward',
    ArrowLeft = 'ArrowLeft',
    ArrowRight = 'ArrowRight',
    ArrowBack = 'ArrowBack',
    ArrowTrendDown = 'ArrowTrendDown',
    ArrowTrendUp = 'ArrowTrendUp',
    ArrowUp = 'ArrowUp',
    ArrowUpward = 'ArrowUpward',
    Athena = 'Athena',
    Attachment = 'Attachment',
    Bot = 'Bot',
    Calendar = 'Calendar',
    Ccp = 'Ccp',
    ChannelChat = 'ChannelChat',
    ChannelEmail = 'ChannelEmail',
    ChannelPhone = 'ChannelPhone',
    ChannelSms = 'ChannelSms',
    ChannelWeb = 'ChannelWeb',
    ChannelUser = 'ChanelUser',
    Chat = 'Chat',
    CheckmarkOutline = 'CheckmarkOutline',
    Clear = 'Clear',
    Close = 'Close',
    Company = 'Company',
    Contacts = 'Contacts',
    Dashboard = 'Dashboard',
    Download = 'Download',
    Delete = 'Delete',
    DeleteCircled = 'DeleteCircled',
    Edit = 'Edit',
    EditCircled = 'EditCircled',
    Email = 'Email',
    Error = 'Error',
    Extension = 'Extension',
    Filter = 'Filter',
    FilterList = 'FilterList',
    Help = 'Help',
    Info = 'Info',
    IncomingEmail = 'IncomingEmail',
    Location = 'Location',
    Menu = 'Menu',
    MoreVert = 'MoreVert',
    MyProfile = 'MyProfile',
    MyStats = 'MyStats',
    Note = 'Note',
    Office365 = 'Office365',
    OutgoingEmail = 'OutgoingEmail',
    PatientChart = 'PatientChart',
    PatientChartV2 = 'PatientChartV2',
    Patients = 'Patients',
    Placeholder = 'Placeholder',
    Play = 'Play',
    Print = 'Print',
    Phone = 'Phone',
    RatingDissatisfied = 'RatingDissatisfied',
    RatingSatisfied = 'RatingSatisfied',
    RatingVerySatisfied = 'RatingVerySatisfied',
    Refresh = 'Refresh',
    Save = 'Save',
    Scripts = 'Scripts',
    Search = 'Search',
    Send = 'Send',
    SignOut = 'SignOut',
    Sms = 'Sms',
    Spam = 'Spam',
    Spinner = 'Spinner',
    Star = 'Star',
    StarOutlined = 'StarOutlined',
    Tickets = 'Tickets',
    View = 'View',
    Warning = 'Warning',
    Web = 'Web',
    LightCheckBoxOff = 'LightCheckBoxOff',
    LightCheckBoxOn = 'LightCheckBoxOn',
    LightRadioOff = 'LightRadioOff',
    LightRadioOn = 'LightRadioOn',
    Comment = 'Comment',
    CwcLogo = 'CwcLogo',
    SentimentSatisfied = "SentimentSatisfied",
    Format = "Format",
    CheckMark = "CheckMark",
    HelioLogo = 'HelioLogo',
    Templates = "Templates",
    ErrorFilled = "ErrorFilled",
    AwsConnect = "AwsConnect",
    Users = "Users",
    Blacklist = "Blacklist",
    Configurations = "Configurations",
    Pause = "Pause",
    CallOutbound = "CallOutbound",
    CallInbound = "CallInbound ",
    CallMissedOutgoing = "CallMissedOutgoing",
    Emergency = "Emergency",
    UserUnknown = "UserUnknown",
    JpgMime = "Jpg",
    PdfMime = "Pdf",
    PngMime = "Png",
    TxtMime = "Txt",
    DocMime = "Doc",
    DocXMime = "DocX",
    XlsMime = "Xls",
    XlsXMime = "XlsX",
    ZipMime = "Zip",
    RarMime = "Rar",
    TiffMime = 'Tiff',
    FallbackMime = "ImagePlaceholder",
    RatingDissatisfiedComment = 'RatingDissatisfiedComment',
    RatingSatisfiedComment = 'RatingSatisfiedComment',
    RatingVerySatisfiedComment = 'RatingVerySatisfiedComment',
    Indicator = 'Indicator',
    AddPhone = 'AddPhone'
}
