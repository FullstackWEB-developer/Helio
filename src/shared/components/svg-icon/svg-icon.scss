.icon-small {
  width: 1rem;
  height: 1rem;
}

.icon-medium {
  width: 1.5rem;
  height: 1.5rem;
}

.icon-medium-18 {
  width: 1.688rem;
  height: 1.688rem;
}

.icon-large {
  width: 2rem;
  height: 2rem;
}

.icon-large-40 {
  width: 2.5rem;
  height: 2.5rem;
}

.icon-x-large {
  width: 3rem;
  height: 3rem;
}

.fill-default {
  fill: var(--light-text-high-emphasis);
}

.stroke-default {
  stroke: var(--icon-stroke-high-emphasis);
  stroke-width: 1;
}

.header-inactive-item-icon {
  fill: var(--header-inactive-item-icon);
}

.header-active-item-icon {
  fill: var(--header-active-item-icon);
}

.navigation-inactive-item-icon {
  fill: var(--navigation-inactive-item-icon);
}

.active-item-icon {
  fill: var(--navigation-active-item-icon);
}

.status-bar-inactive-item-icon {
  fill: var(--status-bar-inactive-item-icon);
}

.edit-icon {
  fill: var(--edit-icon);
}

.search-icon {
  fill: var(--search-icon);
}

.channel-icon-fill {
  fill: var(--channel-icon);
}

.channel-icon-stroke {
  stroke: var(--channel-icon);
}

.rgba-05-fill {
  fill: var(--light-text-medium-emphasis);
}
.rgba-03-fill {
  fill: var(--rgba-color-03);
}

.rgba-038-fill {
  fill: var(--rgba-color-038);
}

.rgba-06-fill {
  fill: var(--rgba-color-06);
}

.rgba-062-fill {
  fill: var(--rgba-color-062);
}

.fill-loading {
  fill: var(--loading-icon-fill);
}

.rgba-025-fill {
  fill: var(--svg-Icon-default-disabled-color);
}

.warning-icon {
  fill: var(--color-yellow);
}

.danger-icon {
  fill: var(--color-red);
}

.success-icon {
  fill: var(--snackbar-icon-success-color);
}

.white-icon {
  fill: var(--color-white);
}

.pagination {
  &-active-icon{
    fill: var(--pagination-active-item);
  }
  &-inactive-icon {
    fill: var(--pagination-inactive-item);
  }
}
