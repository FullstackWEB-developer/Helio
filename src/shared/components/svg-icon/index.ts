import SvgIcon from './svg-icon';
import {Icon} from './icon';

export default SvgIcon;
export {Icon};
