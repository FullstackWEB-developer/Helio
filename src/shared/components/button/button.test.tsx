import i18n from '../../../i18n';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import React from 'react';
import Button from '@components/button/button';
import {Icon} from '@components/svg-icon';

describe("Button tests", () => {
    let container: HTMLDivElement | null;
    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders button correctly", async () => {
        const {asFragment} = render(<Button type='button' label='test-button'/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders big button correctly", async () => {
        const {asFragment} = render(<Button buttonType={'big'} label='test-button'/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders secondary-big button correctly", async () => {
        const {asFragment} = render(<Button buttonType={'secondary-big'} label='test-button'/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders secondary button correctly", async () => {
        const {asFragment} = render(<Button buttonType={'secondary'} label='test-button'/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders small button correctly", async () => {
        const {asFragment} = render(<Button buttonType={'small'} label='test-button'/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders secondary-medium button correctly", async () => {
        const {asFragment} = render(<Button buttonType={'secondary-medium'} label='test-button'/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders disabled button correctly", async () => {
        const {asFragment} = render(<Button buttonType={'secondary-medium'} disabled={true} label='test-button'/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders icon button correctly", async () => {
        const {asFragment} = render(<Button buttonType={'secondary-medium'} icon={Icon.Add} label='test-button'/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders loading button correctly", async () => {
        const {asFragment} = render(<Button isLoading={true} buttonType={'secondary-medium'} label='test-button'/>);
        expect(asFragment()).toMatchSnapshot();
    });
})
