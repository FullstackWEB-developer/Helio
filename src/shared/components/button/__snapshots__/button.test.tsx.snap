// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests renders big button correctly 1`] = `
<DocumentFragment>
  <button
    class="big-button"
    type="button"
  >
    <div
      class="flex flex-row space-x-2 justify-center items-center"
    >
      <div>
        test-button
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button tests renders button correctly 1`] = `
<DocumentFragment>
  <button
    class="medium-button"
    type="button"
  >
    <div
      class="flex flex-row space-x-2 justify-center items-center"
    >
      <div>
        test-button
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button tests renders disabled button correctly 1`] = `
<DocumentFragment>
  <button
    class="secondary-medium-button"
    disabled=""
    type="button"
  >
    <div
      class="flex flex-row space-x-2 justify-center items-center"
    >
      <div>
        test-button
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button tests renders icon button correctly 1`] = `
<DocumentFragment>
  <button
    class="secondary-medium-button"
    type="button"
  >
    <div
      class="flex flex-row space-x-2 justify-center items-center"
    >
      <div
        class="h-6 w-6 inline-flex  align-middle "
      >
        <div
          class=""
        >
          <svg
            class="icon-medium "
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
            />
            <path
              class="enabled-icon-fill"
              d="M19,13H13v6H11V13H5V11h6V5h2v6h6Z"
            />
          </svg>
        </div>
      </div>
      <div>
        test-button
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button tests renders loading button correctly 1`] = `
<DocumentFragment>
  <button
    class="secondary-medium-button loading"
    disabled=""
    type="button"
  >
    <div
      class="flex flex-row space-x-2 justify-center items-center"
    >
      <div
        class="flex flex-col space-y-4 items-center justify-center undefined"
      >
        <div>
          <div
            class=""
          >
            <svg
              class="spinner icon-large-40 "
              viewBox="0 0 40 40"
            >
              <circle
                class="path"
                cx="20"
                cy="20"
                fill="none"
                r="16"
                stroke-width="4"
              />
            </svg>
          </div>
        </div>
      </div>
      <div />
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button tests renders secondary button correctly 1`] = `
<DocumentFragment>
  <button
    class="secondary-button"
    type="button"
  >
    <div
      class="flex flex-row space-x-2 justify-center items-center"
    >
      <div>
        test-button
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button tests renders secondary-big button correctly 1`] = `
<DocumentFragment>
  <button
    class="secondary-big-button"
    type="button"
  >
    <div
      class="flex flex-row space-x-2 justify-center items-center"
    >
      <div>
        test-button
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button tests renders secondary-medium button correctly 1`] = `
<DocumentFragment>
  <button
    class="secondary-medium-button"
    type="button"
  >
    <div
      class="flex flex-row space-x-2 justify-center items-center"
    >
      <div>
        test-button
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button tests renders small button correctly 1`] = `
<DocumentFragment>
  <button
    class="small-button"
    type="button"
  >
    <div
      class="flex flex-row space-x-2 justify-center items-center"
    >
      <div>
        test-button
      </div>
    </div>
  </button>
</DocumentFragment>
`;
